core/src/main/scala/dagr/core/execsystem/TopLikeStatusReporter.scala

```diff
@@ -0,0 +1,214 @@
+ * THE SOFTWARE.
+ */
+
+package dagr.core.execsystem
+
+import dagr.commons.util.BiMap
+import dagr.commons.util.StringUtil._
+import dagr.commons.util.TimeUtil._
+import dagr.core.tasksystem.{InJvmTask, ProcessTask, Task, UnitTask}
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.tools.jline_embedded.{Terminal => JLineTerminal, TerminalFactory}
+
+/** Simple methods for a terminal */
+object Terminal {
+
+  /** Gets a new terminal for each call, in case the terminal is re-sized. */
+  private def terminal: JLineTerminal = TerminalFactory.get()
+
+  def width: Int = terminal.getWidth
+
+  def height: Int = terminal.getHeight
+
+  def supportsAnsi: Boolean = terminal.isAnsiSupported
+}
+
+/** ANSI Strings for cursor movements.
+  *
+  * See: http://www.tldp.org/HOWTO/Bash-Prompt-HOWTO/x361.html
+  */
+object CursorMovement {
+  def positionCursor(line: Int, column: Int): String = s"\033[$line;${column}H"
+
+  def moveUp(numLines: Int): String = s"\033[${numLines}A"
+
+  def moveDown(numLines: Int): String = s"\033[${numLines}B"
+
+  def moveRight(numColumns: Int): String = s"\033[${numColumns}C"
+
+  def moveLeft(numColumns: Int): String = s"\033[${numColumns}D"
+
+  def clearScreen: String = "\033[2J"
+
+  def eraseToEOL: String = "\033[K"
+
+}
+
+/** Mix this trait into a [[dagr.core.execsystem.TaskManager]] to provide the `refresh()` method for a
+  * top-like status interface. */
+trait TopLikeStatusReporter {
+  this: TaskManager =>
+
+  /** The header for the report */
+  private def reportHeader: List[String] = List(
+    "ID", "NAME", "STATUS", "STATE", "CORES", "MEMORY",
+    "SUBMISSION_DATE", "START_DATE", "END_DATE",
+    "EXECUTION_TIME", "TOTAL_TIME", "ATTEMPT_INDEX"
+  )
+
+  /** Truncates the line if it is too long and appends "...", otherwise returns the line */
+  private def truncateLine(line: String, maxColumns: Int): String = if (maxColumns < line.length) line.substring(0, line.length-3) + "..." else line
+
+  /** Splits the string by newlines, and truncates each line if necessary.  Enforces a maximum number of lines. */
+  private def wrap(output: String, maxLines: Int, maxColumns: Int): String = {
+    var lines = output.split("\n")
+    lines = if (maxLines <= lines.length) {
+      lines.slice(0, maxLines-2) ++ List("...")
+    }
+    else {
+      lines
+    }
+    lines.map(truncateLine(_, maxColumns)).mkString("\n")
+  }
+
+  /** A row for the report.  Each row is a single task. This should match the column headers returned by [[reportHeader]]. */
+  private def reportRow(taskInfo: TaskExecutionInfo): List[String] = {
+    // get the total execution time, and total time since submission
+    val (executionTime: String, totalTime: String) = TaskExecutionInfo.getDurationSinceStartAndFormat(taskInfo)
+    // The state of execution
+    val graphNodeState = graphNodeStateFor(taskInfo.taskId).map(_.toString).getOrElse("NA")
+    List(
```

Comment:
Do you think it would be worth the added robustness to make the headers/fields an enum (java or scala) and have reportRow return a Map[ReportField, String]?  Would make it much less error prone to re-order, add and remove fields in the future.